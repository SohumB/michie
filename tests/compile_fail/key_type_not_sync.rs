use michie::memoized;
use std::rc::Rc;

fn generic_in_impl() {
    #[memoized(key_expr = input)]
    fn f(input: Rc<()>) -> () {
        ()
    }
}

fn main() {}
