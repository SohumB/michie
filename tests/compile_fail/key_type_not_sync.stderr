error[E0277]: `Rc<()>` cannot be shared between threads safely
 --> tests/compile_fail/key_type_not_sync.rs:5:5
  |
5 |     #[memoized(key_expr = input)]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Rc<()>` cannot be shared between threads safely
  |
  = help: the trait `Sync` is not implemented for `Rc<()>`
note: required by a bound in `obtain_immutable_cache`
 --> tests/compile_fail/key_type_not_sync.rs:5:5
  |
5 |     #[memoized(key_expr = input)]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `obtain_immutable_cache`
  = note: this error originates in the attribute macro `memoized` (in Nightly builds, run with -Z macro-backtrace for more info)
